docroot/modules/custom/sitenow_intranet/sitenow_intranet.module

```diff
@@ -44,18 +44,32 @@ function sitenow_intranet_restrict_ip_access_denied_page_alter(&$page) {
     '#prefix' => '<h1 class="page-title bold-headline--serif bold-headline text-align-center">',
     '#suffix' => '</h1>',
   ];
-  $page['main']['access_denied_message'] = [
-    '#markup' => t('You are not authorized to access this page.'),
-    '#prefix' => '<p class="text-align-center">',
-    '#suffix' => '</p>',
-  ];
-  $page['main']['continuation_message'] = [
-    '#markup' => t('Continue by logging in with your HawkID.'),
-    '#prefix' => '<p class="text-align-center">',
-    '#suffix' => '</p>',
-  ];
+  // Display a different message to authenticated users with no other roles.
+  $roles = \Drupal::currentUser()->getRoles();
+  if (count($roles) === 1 && in_array('authenticated', $roles)) {
+    $page['main']['access_denied_message'] = [
+      '#markup' => t('You do not have sufficient privileges to access this page. Please contact an administrator.'),
```

Comment:
Since the whole point of the intranet is that there is no public-facing part of it, would it make sense to change this into a friendlier "Please login to access this page" message?

Reply:
Yeah, this message is a bit different. A user would encounter this if they have an account on the site but don't have any roles (stripped by being removed from an AD group or not properly added by a webmaster). Line 57 is the start of the anonymous visitor message where it is suggested they login.

Reply:
Got it.